src/common/middleware/call-api.js

```diff
@@ -5,7 +5,7 @@ export const CALL_API = 'CALL_API';
 
```

Comment:
I don't mind having it implemented with promises for sake of debugging.

Maybe it would be worth putting a comment about it into the code itself so if we end up wondering why we don't use async/await here we will have some explanation.